app/controllers/createAccount.js

```diff
@@ -2,16 +2,22 @@ const mongoose = require("mongoose");
 const User = require("../models/User");
 
 function createAccount(req, res, next) {
-    const {username, password, firstname, lastname, bike} = req.body;
+    const {username, password, firstname, lastname, age, gender, preferred_age, preferred_gender} = req.body;
     const newUser = new User({
-        _id: new mongoose.Types.ObjectId(),
-        username: username.toLowerCase(),
-        password: password,
-        firstName: firstname,
-        lastName: lastname,
-        bike: bike,
-        posts: [],
-        reactions: []
+      _id: new mongoose.Types.ObjectId(),
+      username: username,
+      password: password,
+      firstName: firstname,
+      lastName: lastname,
+      profilePicture: null,
+      beers: null,
+      age: age,
+      gender: gender,
+      preferred_age: {
```

Comment:
I think you can delete the preferred age and gender or ask @Chazzers if there's a way you can combine your features together. Otherwise you can delete this because it makes @Chazzers' filter obsolete.